components/turret.py

```diff
@@ -48,19 +37,8 @@ class Turret:
     # The number of cycles that we must be within the error to decide we're done.
     TICKS_TO_SETTLE = 10
 
-    def __init__(self):
-        # Note that we don't know where the turret actually is until we've
-        # seen an index.
-        self.current_azimuth: int
-        self.current_state = self.IDLE
-
     def on_enable(self) -> None:
-        self.motor.stopMotor()
-        self.run_indexing()
-        self.motor.setSelectedSensorPosition(0, 0)
-        self.baseline_azimuth = self.motor.getSelectedSensorPosition(0)
-        self.baseline_is_provisional = True
-        self.current_azimuth = self.baseline_azimuth
+        self.scan()
```

Comment:
I was given to understand that we should always stop the motors on enable and wait for a controller to tell us to do something, as a matter of safety. I think it's a good idea, too, keeping all activity the province of the controller,  with this component just obeying commands. Of course then we need an IDLE state, though I can see that we probably don't need to return to IDLE to indicate that we're ready to fire, and probably don't want to, as we're asking a motor to hold a given position rather than just waiting for a command.

Reply:
Yes, that makes sense. We'll stop the motors on enable.
But I think leaving the controller always tracking a position is the best option. Mechanically, stopping the slewing motor means that the turret can be rotated by the reaction of the spinning shooter flywheels.

Reply:
Sure. As long as we make it explicitly clear in this file that the intention when slewing by command is to slew to a position _and hold it_. Perhaps we should rename the methods slew_and_hold and slew_to_azimuth_and_hold.

Reply:
Also, the turret shouldn't be initiating a scan by itself, so I have removed that in a commit now on this branch.